File: 540-single-element-in-a-sorted-array/single-element-in-a-sorted-array.cpp
class Solution {
public:
    int singleNonDuplicate(vector<int>& nums) {
        // Solution 1

        // int ans = 0;
        // for(auto ele: nums) ans ^= ele;
        // return ans;
        

        // Solution 2

        int l = 0, r = nums.size()-1;
        while(l < r){
            int mid = (l+r)/2;
            if(mid%2 == 1) mid--;
            if(nums[mid+1] == nums[mid]) l = mid+2;
            else r = mid;
        }
        return nums[l];
    }
};

File: 540-single-element-in-a-sorted-array/single-element-in-a-sorted-array.py
class Solution:
    def singleNonDuplicate(self, nums: List[int]) -> int:

        l, r = 0, len(nums)-1

        while(l <= r):
            mid = (l+r)//2
            if (mid+1) < len(nums) and nums[mid] == nums[mid+1]:
                if mid % 2 == 0:
                    l = mid + 1
                else:
                    r = mid - 1
            else:
                if mid % 2 == 1:
                    l = mid + 1
                else:
                    r = mid - 1
        
        return nums[l]

        